### PR TITLE
Exception handling for devices

### PR DESCRIPTION
**Pull request by hdgx**
Added code to file - openvino_execution_provider.cc to break process if the target hardware devices are not present.

### CODE REVIEW

File: onnxruntime/core/providers/openvino/openvino_execution_provider.cc [hunk 1]
```diff
@@ -25,23 +25,22 @@ OpenVINOExecutionProvider::OpenVINOExecutionProvider(const OpenVINOExecutionProv
...
-        break;
-      }
+
+bool device_found = false;
+auto available_devices = openvino_ep::BackendManager::GetGlobalContext().ie_core.GetAvailableDevices();
+  for (auto device : available_devices) {
+    if (device == info.device_type_) {
+      device_found = true;
+      break;
     }
```
Comment (sfatimar) [inline onnxruntime/core/providers/openvino/openvino_execution_provider.cc:35]: Maybe add some comment here on the new behaviour

Comment (hdgx) [reply onnxruntime/core/providers/openvino/openvino_execution_provider.cc:36]: added comments to branch - https://github.com/intel/onnxruntime/blob/hdgx_onnx_devicecheck/
The existing pull request should handle this comments as well when merged

Comment (hdgx) [reply onnxruntime/core/providers/openvino/openvino_execution_provider.cc:36]: commit details - https://github.com/intel/onnxruntime/pull/125/commits/7439d309c6696cfc04a402ef1e517a038b7ddca5

**Comment by sfatimar** [comment]
RTOM

**Pull request merged**